### DISCUSSION TITLE
Proc macros stopped unwrapping after updating rustup toolchain to 1.70

### DISCUSSION THREAD

**Discussion by cidit** [helix-editor/helix#7476] [Troubleshooting] [answered]
### Summary

title says all.
i have a feeling im missing something very obvious.
 the exact error is this: 
```proc macro `<name of macro>` not expanded: Proc-macro dylib loading failed: unsupported metadata version 7 (unresolved-proc-macro)```

### Reproduction Steps

 

[Recording](https://asciinema.org/a/ljuwPnGpCoTI9FsLinw1A2Moz)
=============================================

I tried this:

1. `hx .`
2. navigated to main.rs
3. opened diagnostics with `Space+d` 

I expected this to happen:

see no errors

Instead, this happened:

saw my proc macros couldnt get unwrapped due to some metadata error.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-06-28T12:53:50.938 helix_view::editor [ERROR] editor error: Register [@] empty
2023-06-28T12:54:09.927 helix_view::editor [ERROR] editor error: no such command: 'ew'
2023-06-28T13:01:19.419 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-06-28T13:03:21.822 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-06-28T13:17:03.970 helix_view::editor [ERROR] editor error: no such command: 'lspr'
2023-06-28T13:17:08.771 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-06-28T13:18:46.468 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-06-28T13:25:52.369 helix_term::application [ERROR] Timed out waiting for language servers to shutdown
```

</details>

### Platform

linux

### Terminal Emulator

alacritty

### Helix Version

helix 23.05 (7f5940be)

**Comment by gabydd** [comment]
You probably need to update rust-analyzer

**Reply by cidit** [reply]
i thought i did by updating the toolchain, and then later tried to update it specifically with `rustup component add rust-analyzer` and it was up to date.

i just did it again to make sure i wasnt spouting bullshit and this was the output:
`info: component 'rust-analyzer' for target 'x86_64-unknown-linux-gnu' is up to date`

**Comment by pascalkuthe** [comment]
proc macros are not handled by helix but RA (as with all language-specific functionality) so this is not an issue with helix

**Reply by cidit** [reply]
to me, it looks like helix is looking for the old version of rust analyser whitch doesnt match the toolchain im using. im thinking that bc when i manually `cargo check` in the command line i dont get the errors, theyre only in helix.

does that make sense? trying to wrap my head around how everything is connected together. ^^"

**Reply by the-mikedavis** [reply]
Helix uses the `rust-analyzer` on your PATH so you can check the version with `rust-analyzer --version`

**Comment by cidit** [comment, marked as answer]
i managed to fix my issue. for some reason, i had a standalone version of rust-analyzer in my path that took priority over the toolchain one.
